Automattic/jetpack
# Jetpack AI: Expose current period usage on feature endpoint

Part of #33566. ~Depends on D125263-code~.

## Proposed changes:

* Add the `current-period-requests-count` property on the `wpcom/v2/sites/SITE/jetpack-ai/ai-assistant-feature` endpoint so we can use it to display usage and trigger upgrade messages

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply this patch using the `jetpack-downloader` command available below (or build and rsync it to your sandbox)
* Sandbox the `public-api` so the requests to the API will be routed to your sandbox
* Go to the WordPress.com Console (https://developer.wordpress.com/docs/api/console/)
* Look for the `wpcom/v2` `/sites/$wpcom_site/jetpack-ai/ai-assistant-feature` request:

<img width="600" alt="Screenshot 2023-10-16 at 18 25 06" src="https://github.com/Automattic/jetpack/assets/6760046/cd3fed8a-5cac-4efc-9557-e550bb0679b3">

* Use your test site as the test subject
* Perform the request and confirm the response contains the new `current-period-requests-count` field:

<img width="600" alt="Screenshot 2023-10-16 at 18 50 32" src="https://github.com/Automattic/jetpack/assets/6760046/cbef4f01-03ce-46b4-91d1-85604ee55e56">

* Confirm that the number is provided in real-time by executing a new AI Assistant query and requesting the endpoint again; it should be incremented in real-time
* Confirm that the provided number is correct; you can find details on how to do it on the D125263-code testing instructions (short version: query the usage tables using `gpr` command inside the `wpsh` shell)
